packages/flutter/lib/src/material/elevation_overlay.dart

```diff
@@ -62,10 +67,27 @@ class ElevationOverlay {
   ///    specifies the exact overlay values for a given elevation.
   static Color overlayColor(BuildContext context, double elevation) {
     final ThemeData theme = Theme.of(context);
+    return _overlayColor(theme.colorScheme.onSurface, elevation);
+  }
+
+  /// Returns a color blended by laying a semi-transparent overlay (using the
+  /// [overlay] color) on top of a surface (using the [surface] color).
+  ///
+  /// The opacity of the overlay depends on [elevation]. As [elevation]
+  /// increases, the opacity of will also increase.
+  ///
+  /// See https://material.io/design/color/dark-theme.html#properties.
+  static Color colorWithOverlay({required Color surface, required Color overlay, required double elevation}) {
+    return Color.alphaBlend(_overlayColor(overlay, elevation), surface);
+  }
+
+  /// Applies the opacity from the dark theme elevation overlay formula at
```

Comment:
I don't understand this comment

Reply:
Ah it was out of date. I simplified this comment because this is a private method anyways. It is simpler and more accurate now. Thanks for spotting that.

Reply:
SGTM!